### PR TITLE
[MRG+1] Reduce the number of warnings with annotations.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4142

### CODE REVIEW

**Comment by Eric89GXL** [comment]
This gets rid of the warnings. LGTM +1 for merge

File: mne/io/base.py [hunk 1]
```diff
@@ -658,7 +658,7 @@ def filenames(self):
         return tuple(self._filenames)
 
     @annotations.setter
-    def annotations(self, annotations):
+    def annotations(self, annotations, emit_warning=True):
```
Comment (agramfort) [inline mne/io/base.py]: why adding a param to toggle warning? we use global settings for this usually.

**Comment by Eric89GXL** [comment]
You mean verbose? I guess we could try to do it here but I'm not sure it
would be any cleaner

File: mne/io/base.py (hunk 1 above)
Comment (jaeilepp) [inline mne/io/base.py]: This is for special cases like reading split files and dealing with floating point inaccuracies. Other option would be to set the private attribute, but then the checks done here should be done elsewhere. IMO this is cleaner.

Comment (agramfort) [inline mne/io/base.py]: @jaeilepp can you tell me where we have a param to decide if warning should be triggered or not?

Comment (Eric89GXL) [inline mne/io/base.py]: There is no user-exposed param, it's just for internal calls like `raw.crop()` that should never emit such warnings (i.e., of course if you crop you will lose data + annotations)

Comment (dengemann) [inline mne/io/base.py]: @Eric89GXL what do you mean? This is a new param in a public function, no?

Comment (jaeilepp) [inline mne/io/base.py]: It's a setter. This gets called by ``raw.annotations = annotations``. This is only exposed if for some reason the user wants to call it with ``mne,BaseRaw.annotations.fset(...)`` (which they shouldn't do).

Comment (agramfort) [inline mne/io/base.py]: got it thanks !

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**